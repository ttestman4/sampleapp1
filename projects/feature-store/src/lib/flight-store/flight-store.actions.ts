import { Action } from '@ngrx/store';
import * as FlightModels from './flight-store.models';
export enum FlightActionTypes {
    Search = '[Flight Store] Search',
    SearchSuccess = '[Flight Store] Search Success',
    SearchFailuer = '[Flight Store] Search Failuer',
    UpdateSearchCriteria = '[Flight Store] Update Search Criteria',
}

/**
 * Every action is comprised of at least a type and an optional
 * payload. Expressing actions as classes enables powerful
 * type checking in reducer functions.
 *
 * See Discriminated Unions: https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions
 */
export class Search implements Action {
    readonly type = FlightActionTypes.Search;
    constructor() { }
}

export class SearchSuccess implements Action {
    readonly type = FlightActionTypes.SearchSuccess;
    constructor(public payload: FlightModels.Result) { }
}
export class SearchFailuer implements Action {
    readonly type = FlightActionTypes.SearchFailuer;
    constructor() { }
}

export class UpdateSearchCriteria implements Action {
    readonly type = FlightActionTypes.UpdateSearchCriteria;
    constructor(public payload: FlightModels.Criteria) { }
}


/**
 * Export a type alias of all actions in this action group
 * so that reducers can easily compose action types
 */
export type FlightActionsUnion =
    Search |
    SearchSuccess |
    SearchFailuer |
    UpdateSearchCriteria;

