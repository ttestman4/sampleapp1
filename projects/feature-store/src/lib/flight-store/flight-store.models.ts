import { Time } from '@angular/common';
import { EntityState } from '@ngrx/entity';

export enum TravelType {
    OneWay = 'One way',
    Return = 'Round trip',
    MultiCity = 'Multi-city',
}
interface FlightDetailsCore {
    origin: string;
    destination: string;
    date: Date;
    travelOrder: number;
}

export interface FlightSearchDetail extends FlightDetailsCore {
    departureAfterTime: Time;
    departureBeforeTime: Time;
}

export interface FlightResultDetail extends FlightDetailsCore {
    name: string;
    departureTime: Time ;
    arrivalTime: Time ;
    duration: Time;
    price: number;
    flightNo: string;
    stops: number;
    multiple?: FlightResultDetail[];
}

export enum PassengerType {
    Adult = 'Adult',
    Child = 'Child',
    InfantInSeat = 'Infant in Seat',
    InfantOnLap = 'Infant on Lap',
}

export interface Passenger {
    type: PassengerType;
    count: number;
}

export enum TravelClass {
    Economy = 'Economy',
    PremiumEconomy = 'Premium Economy',
    Business = 'Business',
    FirstClass = 'FirstClass',
}

export interface Criteria {
    flightSearchDetails: EntityState<FlightSearchDetail>;
    passengers: Passenger[];
    travelType: TravelType;
    travelClass: TravelClass;
    bags: number;
    stops: number;
    price: number;

}

export enum ResultSortBy {
    BestFlights = 'Best Flights',
    Price = 'Price',
    DepartureTime = 'Departure time',
    ArrivalTime = 'Arrival time',
    Duration = 'Duration',
}

export interface Result {
    flightDetails: FlightResultDetail[];
    sortBy: ResultSortBy;
}

export interface SearchState {
    criteria: Criteria;
    result: Result;
}
