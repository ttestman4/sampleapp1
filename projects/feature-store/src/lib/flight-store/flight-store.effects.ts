import { Injectable } from '@angular/core';
import { Actions, Effect, ofType } from '@ngrx/effects';
import { select, Store } from '@ngrx/store';
import { of } from 'rxjs';
import { catchError, map, switchMap, withLatestFrom } from 'rxjs/operators';
import * as FlightActions from './flight-store.actions';
import * as FlightModels from './flight-store.models';
import * as FlightSelectors from './flight-store.selectors';
import { FlightStoreService } from './flight-store.service';

let nextFlightEffectsId = 1;

export function ResetNextFlightEffectsId() {
    nextFlightEffectsId = 1;
}
@Injectable({
    providedIn: 'root'
})
export class FlightEffects {

    @Effect()
    search$ = this.actions$.pipe(
        ofType<FlightActions.Search>(FlightActions.FlightActionTypes.Search),
        withLatestFrom(this.store.pipe(
            select(FlightSelectors.selectCriteria)),
            ((action, criteria) => ({ action, criteria }))
        ),
        switchMap((data) => {
            return this.flightService.search(data.criteria).pipe(
                map((response: FlightModels.Result) => {
                    return new FlightActions.SearchSuccess(response);
                }),
                catchError(_err => {
                    return of(new FlightActions.SearchFailuer());
                }));
        })
    );

    @Effect()
    upsertFlightSearchDetails$ = this.actions$.pipe(
        ofType<FlightActions.UpsertFlightSearchDetails>(FlightActions.FlightActionTypes.UpsertFlightSearchDetails),
        map((_action) => {
            return new FlightActions.Search();
        })
    );

    constructor(
        private actions$: Actions<FlightActions.FlightActionsUnion>,
        private flightService: FlightStoreService,
        private store: Store<FlightModels.SearchState>
    ) {
        const id = nextFlightEffectsId++;
        if (id > 1) {
            throw new Error(
                'FlightEffects is already loaded. Import it in the AppModule only');
        }
    }
}
