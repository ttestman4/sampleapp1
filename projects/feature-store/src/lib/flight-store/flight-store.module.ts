import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { ModuleWithProviders, NgModule, Optional, SkipSelf } from '@angular/core';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';
import { FlightEffects } from './flight-store.effects';
import { reducer } from './flight-store.reducer';

export * from './flight-store.actions';
export * from './flight-store.models';

@NgModule({
  declarations: [],
  imports: [
    CommonModule,
    /**
     * StoreModule.forFeature is used for composing state
     * from feature modules. These modules can be loaded
     * eagerly or lazily and will be dynamically added to
     * the existing state.
     */
    StoreModule.forFeature('flight', reducer),
    /**
     * Effects.forFeature is used to register effects
     * from feature modules. Effects can be loaded
     * eagerly or lazily and will be started immediately.
     *
     * All Effects will only be instantiated once regardless of
     * whether they are registered once or multiple times.
     */
    EffectsModule.forFeature([FlightEffects]),
    HttpClientModule,
  ]
})
export class FlightStoreModule {
  constructor(@Optional() @SkipSelf() parentModule: FlightStoreModule) {
    if (parentModule) {
      throw new Error(
        'FlightStoreModule is already loaded. Import it in the AppModule only');
    }
  }
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: FlightStoreModule,
      providers: [
      ]
    };
  }
  static forChild(): ModuleWithProviders {
    return {
      ngModule: FlightStoreModule,
      providers: [
      ]
    };
  }
  static forTestReset() {
  }
}
