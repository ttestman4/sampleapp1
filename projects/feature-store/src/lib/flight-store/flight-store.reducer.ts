import { createFeatureSelector } from '@ngrx/store';
import * as FlightActions from './flight-store.actions';
import { ResultSortBy, SearchState, TravelClass, TravelType } from './flight-store.models';

export const featureName = 'flight';

export interface State extends SearchState {
    name: string;
}

// export const flightSearchDetailAdapter: EntityAdapter<FlightSearchDetail> = createEntityAdapter<FlightSearchDetail>({
//     selectId: (a) => a.travelOrder,
//     sortComparer: (a, b) => (a.travelOrder - b.travelOrder),
// });

export const initialState: State = {
    name: 'Flight Store',
    criteria: {
        flightSearchDetails: [],
        passengers: [],
        travelType: TravelType.Return,
        travelClass: TravelClass.Economy,
        bags: 1,
        stops: 2,
        price: 0,
    },
    result: {
        flightDetails: [],
        sortBy: ResultSortBy.BestFlights,
    }
};

export function reducer(
    state = initialState,
    action: FlightActions.FlightActionsUnion
): State {
    switch (action.type) {
        case FlightActions.FlightActionTypes.UpdateSearchCriteria: {
            return {
                ...state,
                criteria: action.payload,
            };
        }
        case FlightActions.FlightActionTypes.SearchSuccess: {
            return {
                ...state,
                result: action.payload
            };
        }
        default: {
            return state;
        }
    }
}


export const selectSearchState = createFeatureSelector<SearchState>(featureName);
