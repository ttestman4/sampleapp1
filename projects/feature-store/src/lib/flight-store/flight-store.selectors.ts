import { createSelector } from '@ngrx/store';
import { selectSearchState } from './flight-store.reducer';

export const selectCriteria = createSelector(selectSearchState,
    (searchData) => searchData.criteria
);

export const selectCriteriaFlightDetails = createSelector(selectCriteria,
    (criteria) => criteria.flightSearchDetails
);

export const selectPriceFilter = createSelector(selectCriteria,
    (criteria) => criteria.price
);

export const selectResult = createSelector(selectSearchState,
    (searchData) => searchData.result
);

export const selecResultFlightDetailsListAll = createSelector(selectResult,
    (result) => result.flightDetails
);

export const selectMaxPrice = createSelector(selecResultFlightDetailsListAll,
    (flightDetailsListAll) => {
        return flightDetailsListAll.reduce((accumulator, currentValue) => {
            return currentValue.reduce((innerAccumulator, innerCurrentValue) => {
                return Math.max(innerAccumulator, innerCurrentValue.price);
            }, accumulator);
        }, 0);
    }
);

export const selectResultFlightDetailsListAllFilteredByPrice = createSelector(
    selectPriceFilter, selecResultFlightDetailsListAll,
    (price, flightDetailsListAll) => {
        return flightDetailsListAll.map(flightDetailsList => {
            return flightDetailsList.filter(flightDetails => {
                return price > 0 ? flightDetails.price <= price : true;
            });
        });
    }
);
