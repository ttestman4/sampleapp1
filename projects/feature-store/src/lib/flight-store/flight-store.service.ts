import { Time } from '@angular/common';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import * as FlightModels from './flight-store.models';

let nextFlightStoreServiceId = 1;

export function ResetNextFlightStoreServiceId() {
  nextFlightStoreServiceId = 1;
}

export interface FlightResultDetailFromRestApi {
  origin: string;
  destination: string;
  date: string;
  name: string;
  departureTime: string;
  arrivalTime: string;
  price: number;
  flightNo: string;
}

@Injectable({
  providedIn: 'root'
})
export class FlightStoreService {
  searchUrl = 'assets/advFlightSearch.json';
  // searchUrl = 'assets/search.json';

  search(criteria: FlightModels.Criteria): Observable<FlightModels.Result> {
    // TBD: In real implementation this is going to post call.
    // Need good deployment mechanics to simulat post call.
    return this.http.get<FlightResultDetailFromRestApi[]>(this.searchUrl).pipe(
      map((response) => {
        return response.map((ele) => {
          const departureTime = this.parseTime(ele.departureTime);
          const arrivalTime = this.parseTime(ele.arrivalTime);
          const duration = this.calculateDuration(
            departureTime,
            arrivalTime
          );
          const resultEle: FlightModels.FlightResultDetail = {
            // rectify result values to UI friendly data structure
            origin: ele.origin,
            destination: ele.destination,
            date: new Date(ele.date),
            travelOrder: 0,
            name: ele.name,
            departureTime,
            arrivalTime,
            duration,
            price: ele.price * criteria.passengers,
            flightNo: ele.flightNo,
            stops: 0,
          };
          return resultEle;
        });
      }),
      map((response) => {
        const resultForAllFlightPlans = criteria.flightSearchDetails.map((singleFlightPlan) => {
          const result: FlightModels.FlightResultDetail[] = [];
          // Build list of direct flights
          // Filter records which match the origin and destination criteria and date
          const directFights = response.filter((ele) => {
            return (
              (ele.origin === singleFlightPlan.origin &&
                ele.destination === singleFlightPlan.destination) &&
              ele.date.valueOf() === singleFlightPlan.date.valueOf()
            );
          }).map(ele => {
            return { ...ele, travelOrder: singleFlightPlan.travelOrder };
          });

          let flightsWithOneStop: FlightModels.FlightResultDetail[] = [];

          if (criteria.stops >= 1) {
            // Build list of of flights with 1 stop
            const departureFlightsMatchingOrigin = response.filter((ele) => {
              return (
                (ele.origin === singleFlightPlan.origin &&
                  ele.destination !== singleFlightPlan.destination) &&
                ele.date.valueOf() === singleFlightPlan.date.valueOf()
              );
            });

            const arrivalFlightsMatchingOrigin = response.filter((ele) => {
              return (
                (ele.origin !== singleFlightPlan.origin &&
                  ele.destination === singleFlightPlan.destination) &&
                ele.date.valueOf() === singleFlightPlan.date.valueOf()
              );
            });


            departureFlightsMatchingOrigin.forEach((depart) => {
              const secondLegFlight = arrivalFlightsMatchingOrigin.filter((arrival) => {
                return depart.destination === arrival.origin &&
                  this.timeToMinutes(this.calculateDuration(depart.arrivalTime, arrival.departureTime)) >= 30;
              });

              const firstLegInstanceAndSecondLegListMerged =
                secondLegFlight.map((second) => {
                  const multipleFlightInstance: FlightModels.FlightResultDetail = {
                    origin: depart.origin,
                    destination: second.destination,
                    date: depart.date,
                    travelOrder: singleFlightPlan.travelOrder,
                    name: 'Multiple',
                    departureTime: depart.departureTime,
                    arrivalTime: second.arrivalTime,
                    duration: this.calculateDuration(
                      depart.departureTime,
                      second.arrivalTime
                    ),
                    price: (depart.price + second.price),
                    flightNo: depart.flightNo + ' => ' + second.flightNo,
                    stops: 1,
                    multiple: [
                      { ...depart, travelOrder: singleFlightPlan.travelOrder },
                      { ...second, travelOrder: singleFlightPlan.travelOrder }
                    ]
                  };
                  return multipleFlightInstance;
                });

              flightsWithOneStop = flightsWithOneStop.concat(firstLegInstanceAndSecondLegListMerged);
            });
          }
          return result.concat(directFights, flightsWithOneStop);
        });
        return resultForAllFlightPlans;
      }),
      map((response) => {
        return {
          flightDetails: response,
          sortBy: FlightModels.ResultSortBy.BestFlights
        };
      })
    );
  }

  constructor(private http: HttpClient) {
    const id = nextFlightStoreServiceId++;
    if (id > 1) {
      throw new Error(
        'FlightStoreService is already loaded. Import it in the AppModule only');
    }
  }

  private parseTime(timeInString: string) {
    const separator = timeInString.split(':', 2);
    const time: Time = { hours: 0, minutes: 0 };
    if (separator.length > 0) {
      time.hours = parseInt(separator[0], 10);
    }
    if (separator.length > 1) {
      time.minutes = parseInt(separator[1], 10);
    }
    return time;
  }

  private calculateDuration(time1: Time, time2: Time) {
    const result: Time = { hours: 0, minutes: 0 };
    const date1 = new Date('1995-12-11');
    const date2 = new Date('1995-12-11');
    date1.setUTCHours(time1.hours);
    date1.setUTCMinutes(time1.minutes);
    date2.setUTCHours(time2.hours);
    date2.setUTCMinutes(time2.minutes);
    const diffDate = new Date(date2.valueOf() - date1.valueOf());
    result.hours = diffDate.getUTCHours();
    result.minutes = diffDate.getUTCMinutes();

    return result;
  }

  private timeToMinutes(time: Time) {
    return time.hours * 60 + time.minutes;
  }
}
