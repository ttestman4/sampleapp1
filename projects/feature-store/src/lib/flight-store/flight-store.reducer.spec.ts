import * as FlighActions from './flight-store.actions';
import * as FlightModels from './flight-store.models';
import { initialState, reducer, State } from './flight-store.reducer';
describe('Flight Reducer', () => {
    it('Unknown action should return the initial state', () => {
        const action = {} as any;

        const result = reducer(initialState, action);

        expect(result).toMatchSnapshot();
    });

    it('Unknown action should return the initial state for undefined initial state input', () => {
        const action = {} as any;

        const result = reducer(undefined, action);

        expect(result).toMatchSnapshot();
    });

    let resultFromPreviousStep: State;

    describe('UpdateFlightSearchDetails', () => {
        it('Valid action should return the added FlightSearchDetailsState', () => {
            const flightSearchDetails: FlightModels.FlightSearchDetail = {
                origin: 'BOM',
                destination: 'GOI',
                date: new Date('2018-12-17'),
                travelOrder: 1,
                departureAfterTime: { hours: 9, minutes: 20 },
                departureBeforeTime: { hours: 16, minutes: 19 },
            };
            const action = new FlighActions.UpsertFlightSearchDetails(flightSearchDetails);

            resultFromPreviousStep = reducer(initialState, action);

            expect(resultFromPreviousStep).toMatchSnapshot();
        });

        it('Valid action should return the updated FlightSearchDetailsState', () => {
            const flightSearchDetails: FlightModels.FlightSearchDetail = {
                origin: 'BOM',
                destination: 'DEL',
                date: new Date('2019-11-17'),
                travelOrder: 1,
                departureAfterTime: { hours: 9, minutes: 20 },
                departureBeforeTime: { hours: 16, minutes: 19 },
            };
            const action = new FlighActions.UpsertFlightSearchDetails(flightSearchDetails);

            resultFromPreviousStep = reducer(resultFromPreviousStep, action);

            expect(resultFromPreviousStep).toMatchSnapshot();
        });

        it('Valid action should return the added FlightSearchDetailsState', () => {
            const flightSearchDetails: FlightModels.FlightSearchDetail = {
                origin: 'GOI',
                destination: 'BOM',
                date: new Date('2018-12-17'),
                travelOrder: 2,
                departureAfterTime: { hours: 9, minutes: 20 },
                departureBeforeTime: { hours: 16, minutes: 19 },
            };
            const action = new FlighActions.UpsertFlightSearchDetails(flightSearchDetails);

            resultFromPreviousStep = reducer(resultFromPreviousStep, action);

            expect(resultFromPreviousStep).toMatchSnapshot();
        });
    });

    describe('SearchSuccess', () => {
        it('Valid action should update the result', () => {
            const result: FlightModels.Result = {
                flightDetails: [{
                    origin: 'BOM',
                    destination: 'DEL',
                    date: new Date('2019-11-17'),
                    travelOrder: 1,
                    name: 'indigo',
                    departureTime: { hours: 9, minutes: 20 },
                    arrivalTime: { hours: 16, minutes: 19 },
                    duration: { hours: 7, minutes: 10 },
                    price: 1,
                    flightNo: '6E-123',
                }],
                sortBy: FlightModels.ResultSortBy.BestFlights
            };
            const action = new FlighActions.SearchSuccess(result);

            resultFromPreviousStep = reducer(resultFromPreviousStep, action);

            expect(resultFromPreviousStep).toMatchSnapshot();
        });
    });
});
