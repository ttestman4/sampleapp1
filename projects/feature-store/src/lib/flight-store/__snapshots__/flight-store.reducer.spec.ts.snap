// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Reducer SearchSuccess Valid action should update the result 1`] = `
Object {
  "criteria": Object {
    "bags": 1,
    "flightSearchDetails": Object {
      "entities": Object {
        "1": Object {
          "date": 2019-11-17T00:00:00.000Z,
          "departureAfterTime": Object {
            "hours": 9,
            "minutes": 20,
          },
          "departureBeforeTime": Object {
            "hours": 16,
            "minutes": 19,
          },
          "destination": "DEL",
          "origin": "BOM",
          "travelOrder": 1,
        },
        "2": Object {
          "date": 2018-12-17T00:00:00.000Z,
          "departureAfterTime": Object {
            "hours": 9,
            "minutes": 20,
          },
          "departureBeforeTime": Object {
            "hours": 16,
            "minutes": 19,
          },
          "destination": "BOM",
          "origin": "GOI",
          "travelOrder": 2,
        },
      },
      "ids": Array [
        1,
        2,
      ],
    },
    "passengers": Array [],
    "price": 0,
    "stops": 10,
    "travelClass": "Economy",
    "travelType": "Round trip",
  },
  "name": "Flight Store",
  "result": Object {
    "flightDetails": Array [
      Object {
        "arrivalTime": Object {
          "hours": 16,
          "minutes": 19,
        },
        "date": 2019-11-17T00:00:00.000Z,
        "departureTime": Object {
          "hours": 9,
          "minutes": 20,
        },
        "destination": "DEL",
        "duration": Object {
          "hours": 7,
          "minutes": 10,
        },
        "flightNo": "6E-123",
        "name": "indigo",
        "origin": "BOM",
        "price": 1,
        "travelOrder": 1,
      },
    ],
    "sortBy": "Best Flights",
  },
}
`;

exports[`Flight Reducer Unknown action should return the initial state 1`] = `
Object {
  "criteria": Object {
    "bags": 1,
    "flightSearchDetails": Object {
      "entities": Object {},
      "ids": Array [],
    },
    "passengers": Array [],
    "price": 0,
    "stops": 10,
    "travelClass": "Economy",
    "travelType": "Round trip",
  },
  "name": "Flight Store",
  "result": Object {
    "flightDetails": Array [],
    "sortBy": "Best Flights",
  },
}
`;

exports[`Flight Reducer Unknown action should return the initial state for undefined initial state input 1`] = `
Object {
  "criteria": Object {
    "bags": 1,
    "flightSearchDetails": Object {
      "entities": Object {},
      "ids": Array [],
    },
    "passengers": Array [],
    "price": 0,
    "stops": 10,
    "travelClass": "Economy",
    "travelType": "Round trip",
  },
  "name": "Flight Store",
  "result": Object {
    "flightDetails": Array [],
    "sortBy": "Best Flights",
  },
}
`;

exports[`Flight Reducer UpdateFlightSearchDetails Valid action should return the added FlightSearchDetailsState 1`] = `
Object {
  "criteria": Object {
    "bags": 1,
    "flightSearchDetails": Object {
      "entities": Object {
        "1": Object {
          "date": 2018-12-17T00:00:00.000Z,
          "departureAfterTime": Object {
            "hours": 9,
            "minutes": 20,
          },
          "departureBeforeTime": Object {
            "hours": 16,
            "minutes": 19,
          },
          "destination": "GOI",
          "origin": "BOM",
          "travelOrder": 1,
        },
      },
      "ids": Array [
        1,
      ],
    },
    "passengers": Array [],
    "price": 0,
    "stops": 10,
    "travelClass": "Economy",
    "travelType": "Round trip",
  },
  "name": "Flight Store",
  "result": Object {
    "flightDetails": Array [],
    "sortBy": "Best Flights",
  },
}
`;

exports[`Flight Reducer UpdateFlightSearchDetails Valid action should return the added FlightSearchDetailsState 2`] = `
Object {
  "criteria": Object {
    "bags": 1,
    "flightSearchDetails": Object {
      "entities": Object {
        "1": Object {
          "date": 2019-11-17T00:00:00.000Z,
          "departureAfterTime": Object {
            "hours": 9,
            "minutes": 20,
          },
          "departureBeforeTime": Object {
            "hours": 16,
            "minutes": 19,
          },
          "destination": "DEL",
          "origin": "BOM",
          "travelOrder": 1,
        },
        "2": Object {
          "date": 2018-12-17T00:00:00.000Z,
          "departureAfterTime": Object {
            "hours": 9,
            "minutes": 20,
          },
          "departureBeforeTime": Object {
            "hours": 16,
            "minutes": 19,
          },
          "destination": "BOM",
          "origin": "GOI",
          "travelOrder": 2,
        },
      },
      "ids": Array [
        1,
        2,
      ],
    },
    "passengers": Array [],
    "price": 0,
    "stops": 10,
    "travelClass": "Economy",
    "travelType": "Round trip",
  },
  "name": "Flight Store",
  "result": Object {
    "flightDetails": Array [],
    "sortBy": "Best Flights",
  },
}
`;

exports[`Flight Reducer UpdateFlightSearchDetails Valid action should return the updated FlightSearchDetailsState 1`] = `
Object {
  "criteria": Object {
    "bags": 1,
    "flightSearchDetails": Object {
      "entities": Object {
        "1": Object {
          "date": 2019-11-17T00:00:00.000Z,
          "departureAfterTime": Object {
            "hours": 9,
            "minutes": 20,
          },
          "departureBeforeTime": Object {
            "hours": 16,
            "minutes": 19,
          },
          "destination": "DEL",
          "origin": "BOM",
          "travelOrder": 1,
        },
      },
      "ids": Array [
        1,
      ],
    },
    "passengers": Array [],
    "price": 0,
    "stops": 10,
    "travelClass": "Economy",
    "travelType": "Round trip",
  },
  "name": "Flight Store",
  "result": Object {
    "flightDetails": Array [],
    "sortBy": "Best Flights",
  },
}
`;
